sklearn/preprocessing/_function_transformer.py

```diff
@@ -64,11 +64,31 @@ def fit(self, X, y=None):
             check_array(X, self.accept_sparse)
         return self
 
+    def get_feature_names(self, input_features=None):
+        """
+        Return feature names for output features
+
+        Parameters
+        ----------
+        input_features : list of string, length len(input_features), optional
+            String names for input features if available. By default,
+            None is used.
+
+        Returns
+        -------
+        output_feature_names : list of string, length len(input_features)
+
+        """
+        if input_features is not None:
+            input_features = list(map(lambda input_feature: 'f(' +
```

Comment:
Yeah, good points. 

> a machine-readable description of a feature's provenance (ideally a structured representation)

Are you saying that string formatting is not powerful enough for these use cases, and there is a need for more structured representation (a dict, a special object)? What additional information are you thinking of?

> a human readable (in limited memory!) description of a feature as distinct from a neighbour

In [eli5](https://github.com/TeamHG-Memex/eli5) library instead of feature name lists we used a special object which generates feature names on a fly if they are autogenerated ('pca__x0' as opposed to 'petal width'), in order to avoid keeping it all in memory, and to save CPU time when only a subset of feature names is needed. For example, a common task is to get feature names which correspond to non-zero coefficients in a particular document, or feature names for top N features with largest coefficients. CPU was even more important than memory because there were noticeable delays in interactive usage.